site/src/components/AppLink/AppLink.tsx

```diff
@@ -29,6 +30,11 @@ export const AppLink: FC<AppLinkProps> = ({
   workspace,
   agent,
 }) => {
+  const preferredProxy = usePreferredProxy()
```

Comment:
TBH this context should return `{ regions: [], selectedRegionID: "abcdef-...", preferredPathAppURL: "", preferredWildcardHostname: "" }`

The context can merge with the first region if they're not set on the selected one. This saves each consumer of this context from having to do the ternarys

Reply:
Good idea :+1:

Reply:
Fixed the ternaries